holidays/countries/central_african_republic.py

```diff
@@ -0,0 +1,165 @@
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from gettext import gettext as tr
+
+from holidays.calendars import _CustomIslamicHolidays
+from holidays.calendars.gregorian import JAN, FEB, MAR, APR, MAY, JUN, JUL, AUG, SEP, OCT, NOV, DEC
+from holidays.groups import ChristianHolidays, InternationalHolidays, IslamicHolidays
+from holidays.observed_holiday_base import ObservedHolidayBase
+
+
+class CentralAfricanRepublic(
+    ObservedHolidayBase, ChristianHolidays, InternationalHolidays, IslamicHolidays
+):
+    """Central African Republic holidays.
+
+    References:
+        * [Public holidays in the Central African Republic](https://en.wikipedia.org/wiki/Public_holidays_in_the_Central_African_Republic)
+        * [PUBLIC HOLIDAYS](https://web.archive.org/web/20171215122602/http://www.ais-asecna.org/pdf/gen/gen-2-1/04gen2-1-01.pdf)
+    """
+
+    country = "CF"
+    default_language = "fr"
+    supported_languages = ("en_US", "fr")
+    # %s (estimated).
+    estimated_label = tr("%s (estimé)")
+    # %s (observed, estimated).
+    observed_estimated_label = tr("%s (observé, estimé)")
+    # December 1, 1958: Autonomy within the French Community.
+    start_year = 1959
+
+    def __init__(self, *args, islamic_show_estimated: bool = True, **kwargs):
+        """
+        Args:
+            islamic_show_estimated:
+                Whether to add "estimated" label to Islamic holidays name
+                if holiday date is estimated.
+        """
+        ChristianHolidays.__init__(self)
+        InternationalHolidays.__init__(self)
+        IslamicHolidays.__init__(
+            self, cls=CentralAfricanRepublicIslamicHolidays, show_estimated=islamic_show_estimated
+        )
+        super().__init__(*args, **kwargs)
+
+    def _populate_public_holidays(self):
+        # New Year's Day.
+        self._add_new_years_day(tr("Jour de l'an"))
+
+        # Barthélemy Boganda Day.
+        self._add_holiday_mar_29(tr("Journée Barthélemy Boganda"))
+
+        # Easter Monday.
+        self._add_easter_monday(tr("Lundi de Pâques"))
+
+        # Labor Day.
+        self._add_labor_day(tr("Fête du Travail"))
+
+        # Ascension Day.
+        self._add_ascension_thursday(tr("Ascension"))
+
+        # Whit Monday.
+        self._add_whit_monday(tr("Pentecôte"))
+
+        if self._year >= 2007:
+            # General Prayer Day.
+            self._add_holiday_jun_30(tr("Journée de prière générale"))
+
+        if self._year >= 1960:
+            # Independence Day.
+            self._add_holiday_aug_13(tr("Jour de l'indépendance"))
+
+        # Assumption Day.
+        self._add_assumption_of_mary_day(tr("Assomption"))
+
+        # All Saints' Day.
+        self._add_all_saints_day(tr("Toussaint"))
+
+        # National Day.
+        self._add_holiday_dec_1(tr("Fête nationale"))
```

Comment:
[National Day was temporarily moved in 1977-1978 and restored back to DEC, 1 in 1979 when Bokassa was removed from power](https://[web.archive.org/web/20250122135347/https://corbeaunews-centrafrique.org/lindependance-et-la-fete-nationale-deux-dates-distinctes-pour-la-republique-centrafricaine/](https://web.archive.org/web/20250122135347/https://corbeaunews-centrafrique.org/lindependance-et-la-fete-nationale-deux-dates-distinctes-pour-la-republique-centrafricaine/))
```suggestion
        # National Day.
        name = tr("Fête nationale")
        if self._year in {1977, 1978}:
            self._add_holiday_dec_4(name)
        else:
            self._add_holiday_dec_1(name)
```